### PR TITLE
REFACTOR: convert invites controller specs to requests

### PR DESCRIPTION
**Pull request by OsamaSayegh**

### CODE REVIEW

File: spec/requests/invites_controller_spec.rb [hunk 1]
```diff
@@ -37,4 +36,385 @@
...
+            expect(json["errors"]["password"]).to be_present
+          end
+        end
+
+        context '.post_process_invite' do
+          before do
+            SiteSetting.queue_jobs = true
+            #Invite.any_instance.stubs(:redeem).returns(user)
+            #Jobs.expects(:enqueue).with(:invite_email, has_key(:invite_id))
+            #user.password_hash = nil
```
Comment (tgxworld) [inline spec/requests/invites_controller_spec.rb]: Are these comments still required?

File: spec/requests/invites_controller_spec.rb [hunk 2]
```diff
@@ -37,4 +36,385 @@
...
+              :user_logged_in, :user_first_logged_in
+            )
+            invite.reload
+            expect(response).to be_success
+            expect(session[:current_user_id]).to eq(invite.user_id)
+            expect(invite.redeemed?).to be_truthy
+          end
+
+          it 'redirects to the first topic the user was invited to' do
+            put "/invites/show/#{invite.invite_key}.json"
```
Comment (tgxworld) [inline spec/requests/invites_controller_spec.rb]: Ah it looks like we didn't test the response status in the past for this test.

File: spec/requests/invites_controller_spec.rb [hunk 3]
```diff
@@ -37,4 +36,385 @@
...
+      end
+    end
+
+    context 'new registrations are disabled' do
+      let(:topic) { Fabricate(:topic) }
+      let(:invite) { topic.invite_by_email(topic.user, "iceking@adventuretime.ooo") }
+      before { SiteSetting.allow_new_registrations = false }
+
+      it "doesn't redeem the invite" do
+        put "/invites/show/#{invite.invite_key}.json"
```
Comment (tgxworld) [inline spec/requests/invites_controller_spec.rb]: missing response code here

File: spec/requests/invites_controller_spec.rb [hunk 4]
```diff
@@ -37,4 +36,385 @@
...
+      end
+    end
+
+    context 'user is already logged in' do
+      let(:topic) { Fabricate(:topic) }
+      let(:invite) { topic.invite_by_email(topic.user, "iceking@adventuretime.ooo") }
+      let!(:user) { sign_in(Fabricate(:user)) }
+
+      it "doesn't redeem the invite" do
+        put "/invites/show/#{invite.invite_key}.json", params: { id: invite.invite_key }
```
Comment (tgxworld) [inline spec/requests/invites_controller_spec.rb]: missing response code here

**Comment by tgxworld** [comment]
@OsamaSayegh Some minor comments but other wise this looks good to me

**Comment by OsamaSayegh** [comment]
@tgxworld PR updated 👍  Mind taking another quick look?

**Pull request merged**